server/proxy/httpproxy/director.go

```diff
@@ -33,20 +34,23 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
-func newDirector(lg *zap.Logger, urlsFunc GetProxyURLs, failureWait time.Duration, refreshInterval time.Duration, stop <-chan struct{}, donec chan<- struct{}) *director {
+func newDirector(lg *zap.Logger, urlsFunc GetProxyURLs, failureWait time.Duration, refreshInterval time.Duration) *director {
 	if lg == nil {
 		lg = zap.NewNop()
 	}
 	d := &director{
 		lg:          lg,
 		uf:          urlsFunc,
 		failureWait: failureWait,
+		stopc:       make(chan struct{}),
+		donec:       make(chan struct{}),
 	}
+	osutil.RegisterInterruptHandler(func() {
+		close(d.stopc)
```

Comment:
I don't think it would be problematic, but looking for input. The tests also close `stopc` - should this handler accommodate the channel already being closed?